### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by Chandra-Pratap-Singh**
Please review and merge task-3

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -3,10 +3,30 @@ import { Routes, RouterModule } from "@angular/router";
...
+import { LoginguardGuard } from "./loginguard.guard";
+import { AdminGuard } from "./admin.guard";
+import { NewsArticleComponent } from "./news-article/news-article.component";
+import { NewsArticleGuard } from "./news-article.guard";
 
 const routes: Routes = [
-  { path: "", component: MainSectionComponent },
-  { path: "addArticle", component: AddArticleSectionComponent },
+  { path: "", component: LoginComponent },
+  // { path: '', component: CommentsComponent },
```
Comment (itaphuviinnay) [inline src/app/app-routing.module.ts]: Remove this un-used / commented statement

File: src/app/comments/comments.component.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Component, OnInit } from '@angular/core';
+import { CommentsService } from '../comments.service';
+
+@Component({
+  selector: 'app-comments',
+  templateUrl: './comments.component.html',
+  styleUrls: ['./comments.component.css']
+})
+export class CommentsComponent implements OnInit {
+  constructor(private commentsSource: CommentsService) { }
```
Comment (itaphuviinnay) [inline src/app/comments/comments.component.ts]: `commentsSource` should be renamed to `commentsService`. Also format this file

File: src/app/header-section-one/header-section-one.component.ts [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
   selector: "app-header-section-one",
   templateUrl: "./header-section-one.component.html",
   styleUrls: ["./header-section-one.component.css"]
 })
 export class HeaderSectionOneComponent implements OnInit {
-  constructor() {}
+  user: any;
 
-  ngOnInit() {}
+  constructor(private userDetails: UserService) {}
```
Comment (itaphuviinnay) [inline src/app/header-section-one/header-section-one.component.ts]: `userDetails` can be renamed to `userDetailsService`

File: src/app/header-section-three/header-section-three.component.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
   <div class="col-md-6 filter-panel text-center">
     <app-filter-panel></app-filter-panel>
   </div>
-  <section class="col-md-3 article-section">
+  <section class="col-md-3 article-section" *ngIf="checkAccess()">
```
Comment (itaphuviinnay) [inline src/app/header-section-three/header-section-three.component.html]: While you can call a typescript function within in the `*ngIf`, I would suggest to have a variable `isAdmin` on the component level and use it here. The variable `isAdmin` will be more meaningful within the `*ngIf` condition

On an other note, calling typescript functions from the html isn't a best practice

File: src/app/header-section-three/header-section-three.component.ts [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
-
-  ngOnInit() {}
+  constructor(private userService: UserService) {}
+  user: {"loginAs":string ,"id":string,"login":boolean};
+  ngOnInit() {
+    this.user=this.userService.getUser();
+  }
+  checkAccess(){
+    return this.userService.getLoginAs() ==="admin";
+  }
```
Comment (itaphuviinnay) [inline src/app/header-section-three/header-section-three.component.ts]: Format this file and see my comment above for `header-section-three.component.html`

File: src/app/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import { UserService } from '../user.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+  loginDetails={loginAs:"",id:"",password:"",login:true};
+  user:{loginAs:string, id:string, login:boolean};
```
Comment (itaphuviinnay) [inline src/app/login/login.component.ts]: You can create an interface for `User` something like below

```
export interface User {
       id: string; 
       loginAs: string;
       login: boolean;
}
```

You can re-use this interface at multiple places and can be changed easily if more fields were to be added / removed

File: src/app/admin.guard.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  providedIn: 'root'
+})
+export class AdminGuard implements CanActivate {
+  constructor(private userService: UserService, private router: Router) {}
+  canActivate(): boolean{
+    if(this.userService.checkLocalStorage() && this.userService.getLoginAs() ==="admin")
+      return true;
+    else
+    {
+      this.router.navigateByUrl('news');
```
Comment (itaphuviinnay) [inline src/app/admin.guard.ts]: You need to add `return false;` statement after this statement

File: src/app/news-article.guard.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NewsArticleGuard implements CanActivate {
+  constructor(private newsService: NewsService, private router: Router){}
+  canActivate(): boolean{
+    if(this.newsService.newsArticle) return true;
+    else {
+      // console.log(this.newsService.newsArticle);
```
Comment (itaphuviinnay) [inline src/app/news-article.guard.ts]: Remove `console.log` statements

File: src/app/news-section/news-section.component.ts [hunk 9]
```diff
@@ -1,39 +1,51 @@
...
 import { NewsService } from "../news.service";
+import { Router } from '@angular/router';
 
 @Component({
   selector: "app-news-section",
   templateUrl: "./news-section.component.html",
   styleUrls: ["./news-section.component.css"]
 })
 export class NewsSectionComponent implements OnInit {
+  readMore: Boolean;
```
Comment (itaphuviinnay) [inline src/app/news-section/news-section.component.ts]: `Boolean` should be `boolean`

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (Chandra-Pratap-Singh) [reply src/app/app-routing.module.ts]: Done

File: src/app/comments/comments.component.ts (hunk 2 above)
Comment (Chandra-Pratap-Singh) [reply src/app/comments/comments.component.ts]: Done

File: src/app/header-section-one/header-section-one.component.ts (hunk 3 above)
Comment (Chandra-Pratap-Singh) [reply src/app/header-section-one/header-section-one.component.ts]: Done

File: src/app/header-section-three/header-section-three.component.html (hunk 4 above)
Comment (Chandra-Pratap-Singh) [reply src/app/header-section-three/header-section-three.component.html]: Done

File: src/app/header-section-three/header-section-three.component.ts (hunk 5 above)
Comment (Chandra-Pratap-Singh) [reply src/app/header-section-three/header-section-three.component.ts]: Done

File: src/app/login/login.component.ts (hunk 6 above)
Comment (Chandra-Pratap-Singh) [reply src/app/login/login.component.ts]: Done

File: src/app/admin.guard.ts (hunk 7 above)
Comment (Chandra-Pratap-Singh) [reply src/app/admin.guard.ts]: done

File: src/app/news-article.guard.ts (hunk 8 above)
Comment (Chandra-Pratap-Singh) [reply src/app/news-article.guard.ts]: done

File: src/app/news-section/news-section.component.ts (hunk 9 above)
Comment (Chandra-Pratap-Singh) [reply src/app/news-section/news-section.component.ts]: done

**Pull request merged**